pageserver/src/layered_repository/layer_map.rs

```diff
@@ -122,11 +122,7 @@ impl LayerMap {
     pub fn get_open(&self, tag: &SegmentTag) -> Option<Arc<InMemoryLayer>> {
         let segentry = self.segs.get(tag)?;
 
-        if let Some(open) = &segentry.open {
-            Some(Arc::clone(open))
-        } else {
-            None
-        }
+        segentry.open.as_ref().map(|open| Arc::clone(open))
```

Comment:
I think the `if let` construct is more readable. At least to my untrained eye, the map() syntax takes a while to parse. It makes me think there's iteration happening but it's just an Option rather than a Collection. And the `as_ref()` adds noise.

Reply:
> It makes me think there's iteration happening

Your intuition is correct: iteration is exactly what's happening here. We're iterating over a collection comprised of _no more than_ one element. Unfortunately, Rust is unable to define a common trait for mappable types (Option, Result, Vec, HashMap, the list goes on), because its type system has no notion of higher-kinded types.

Personally, I'd go as far as to write the following:

```rust
segentry.open.as_ref().map(Arc::clone)
```

> And the as_ref() adds noise

This is subjective. As a matter of fact, `as_ref` allows one to chain calls when a subsequent method requires a reference to the data.

Reply:
I'm personally leaning towards @funbringer's suggestion, for me map looks pretty organic here

Reply:
Ok, I'm outvoted by you, @funbringer and Clippy then :-).